implementations/micrometer-registry-datadog/src/main/java/io/micrometer/datadog/DatadogMeterRegistry.java

```diff
@@ -221,21 +352,68 @@ private void addToMetadataList(Map<String, DatadogMetricMetadata> metadata, Mete
             return; // we can't set metadata correctly without the application key
 
         Meter.Id fullId = id;
-        if (suffix != null)
+        if (suffix != null) {
             fullId = idWithSuffix(id, suffix);
+        }
 
         String metricName = getConventionName(fullId);
         if (!verifiedMetadata.contains(metricName)) {
             metadata.put(metricName, new DatadogMetricMetadata(fullId, stat, config.descriptions(), overrideBaseUnit));
         }
     }
 
+    // VisibleForTesting
+    void writeMetricViaStatsd(Meter.Id id, @Nullable String suffix, double value, Statistic statistic,
+            @Nullable String overrideBaseUnit) {
+        if (statsDClient == null) {
+            return;
+        }
+
+        Meter.Id fullId = id;
+        if (suffix != null) {
+            fullId = idWithSuffix(id, suffix);
+        }
+
+        Iterable<Tag> tags = getConventionTags(fullId);
+
+        List<String> tagsList = tags.iterator().hasNext() ? stream(tags.spliterator(), false)
+            .map(t -> "\"" + escapeJson(t.getKey()) + ":" + escapeJson(t.getValue()) + "\"")
+            .collect(toList()) : new ArrayList<String>();
+        if (config.hostTag() != null && !Objects.equals(config.hostTag(), "")) {
+            tagsList.add("host:" + config.hostTag());
```

Comment:
The value of `config.hostTag()` is the name of the Micrometer tag to get the Datadog `host` tag value from. See https://github.com/micrometer-metrics/micrometer/blob/3d7a3901c68df189f185a931cee89f33c188d461/implementations/micrometer-registry-datadog/src/main/java/io/micrometer/datadog/DatadogMeterRegistry.java#L243-L247